mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java

```diff
@@ -1357,7 +1376,30 @@ private static class MCRXslImportResolver implements URIResolver {
 
         @Override
         public Source resolve(String href, String base) throws TransformerException {
-            String importXSL = MCRXMLFunctions.nextImportStep(href.substring(href.indexOf(':') + 1));
+            final String baseURI = getParentDirectoryResourceURI(base);
+            // set xslt folder
+            final String xslFolder;
+            if (StringUtils.contains(baseURI, "/xsl/")) {
```

Comment:
baseURI.startswith would work more secure

Reply:
If I understand this correctly, getParentDirectoryResourceURI always returns a string starting with resource (https://github.com/MyCoRe-Org/mycore/blob/2022.06.x/mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java#L293)
e.g.:
base: jar:file:/home/musiclpz/production/build/webapps/WEB-INF/lib/mycore-solr-2023.08-SNAPSHOT.jar!/xslt/solr/indexing/mycoreobject-dynamicfields.xsl
baseuri: resource:xslt/solr/indexing/

so startsWith resource:xsl/ should be feasible.

However, in some edge cases getParentDirectoryResourceURI returns the given base. In these cases, this would not work. But I am not totally aware if in these cases something else goes wrong. If we can ignore these edge cases, we can test on startsWith resource:xsl/

Reply:
> If I understand this correctly, getParentDirectoryResourceURI always returns a string starting with resource (https://github.com/MyCoRe-Org/mycore/blob/2022.06.x/mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java#L293)
e.g.:
base: jar:file:/home/musiclpz/production/build/webapps/WEB-INF/lib/mycore-solr-2023.08-SNAPSHOT.jar!/xslt/solr/indexing/mycoreobject-dynamicfields.xsl
baseuri: resource:xslt/solr/indexing/

Yes you are right.

> However, in some edge cases getParentDirectoryResourceURI returns the given base. In these cases, this would not work. But I am not totally aware if in these cases something else goes wrong. If we can ignore these edge cases, we can test on startsWith resource:xsl/

I think in the edge cases the property should be used. So check for startsWith resource:xsl/ or resource:xslt/ and if both false than use the property.